aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -441,6 +445,29 @@ mod loader {
             self
         }
 
+        /// When true, send this request to the FIPS-compliant regional endpoint.
+        ///
+        /// If the configured endpoint does not have a FIPS compliant endpoint, dispatching
+        /// the request will return an error.
+        ///
+        /// **Note**: Not all services and regions support FIPS. If a service does not support FIPS,
+        /// this setting will have no effect.
```

Comment:
```suggestion
        /// When true, send this request to the FIPS-compliant regional endpoint.
        ///
        /// If the configured endpoint it not a FIPS-compliant endpoint, dispatching
        /// the request will return an error.
        ///
        /// **Note**: Not all services and regions support FIPS. If a service does not support FIPS,
        /// this setting will have no effect.
```

I'm having a bit of trouble understanding which cases will return an error. Is this table correct?

| `use_fips = <bool>` | service supports FIPS | will error |
| -------------------- | ---------------------- | --------- |
| true | yes | no |
| true | no | yes |
| false | yes | no |
| false | no | no|

What is the difference between a configured endpoint being FIPS compliant vs a service supporting FIPS?